src/Plugin/TfaValidation/TfaTotp.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/**
+ * @file TfaTOTP class
+ */
+
+namespace Drupal\tfa\Plugin\TfaValidation;
+
+use Drupal\tfa\Plugin\TfaBasePlugin;
+use Drupal\tfa\Plugin\TfaValidationInterface;
+use Drupal\tfa\Plugin\TfaSetupInterface;
+use Drupal\Core\Url;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\user\Entity\User;
+use Drupal\Component\Utility\Crypt;
+use Drupal\Core\Site\Settings;
+use PHPGangsta_GoogleAuthenticator as GoogleAuthenticator;
```

Comment:
You can use this library in ga login.
Just need to use the plugin id to reference it and then use it's methods for validation.
We are doing this so that other plugins don't have to depend on GA Login to make use of such libraries like the TOTP and HOTP